### PR TITLE
Feat: Resource share functionality

### LINKED ISSUE #58: feat(resource-tool): share resource
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**
Create a `resource-share` component that shows a share button. When tapped the component should prompt resource sharing via the [Capacitor Share API](https://capacitorjs.com/docs/apis/share) API

**Additional context**
Recommend a simple share button that appears alongside file resources, component code at `apps\picsa-tools\resources-tool\src\app\components\resource-item\file`
image

On web only URLs are supported for sharing, so if using web recommend just sharing the current URL instead of resource

On android we want to share the native file, although developer will need to check whether the file can be shared using the URI provided for resource files or whether they will first need to be written to a temporary storage location before sharing

### PR DESCRIPTION
**Pull request by khalifan-kfan**
## Description

This feature enables a user to share a resource to others on web and on native devices.

## Discussion

Following issue 
closes #58

## Screenshots / Videos
**Update 2024-06-19** - Share file on native device
_Note_ - after sharing to email the file appears in email attachment (not shown). Similar behaviour would be seen if sharing the file on whatsapp, bluetooth or similar

[untitled.webm](https://github.com/e-picsa/picsa-apps/assets/10515065/51c121ab-01d9-422f-b710-cb26e8dca8fc)

**Original**
Screenshot 2024-05-27 at 07 53 41

### CODE REVIEW

**Review by chrismclarke — CHANGES_REQUESTED** [review]
Looks like definitely headed in the right direction, see a few minor (non-blocking) comments inline and then the one blocking about sharing as an actual file and not just links.

**Comment by khalifan-kfan** [comment]
Thanks for the feedback @chrismclarke, Let me look into fixing the outlined issues.

File: apps/picsa-tools/resources-tool/src/app/services/resources-tool.service.ts [hunk 1]
```diff
@@ -228,4 +233,44 @@ export class ResourcesToolService extends PicsaAsyncService {
...
+          title: 'Share Resource',
+          url: url,
+          dialogTitle: 'Share Resource Link',
+        });
+      } catch (error) {
+        console.error(error);
+      }
+    } else {
+      try {
+        //api is not compatible with some browser versions
```
[the remark this answers is not in the document]
Comment (khalifan-kfan) [reply apps/picsa-tools/resources-tool/src/app/services/resources-tool.service.ts:283]: Apparently my chrome browser couldn't either but my safari could

File: apps/picsa-tools/resources-tool/src/app/services/resources-tool.service.ts [hunk 2]
```diff
@@ -228,4 +233,44 @@ export class ResourcesToolService extends PicsaAsyncService {
...
+    if (resourceType === 'link' && resource?.subtype === 'play_store') {
+      url = `https://play.google.com/store/apps/details?id=${url}`;
+    }
+    if (Capacitor.isNativePlatform()) {
+      try {
+        await Share.share({
+          title: 'Share Resource',
+          url: url,
+          dialogTitle: 'Share Resource Link',
+        });
```
[the remark this answers is not in the document]
Comment (khalifan-kfan) [reply apps/picsa-tools/resources-tool/src/app/services/resources-tool.service.ts:246]: I may need some help on this part @chrismclarke

**Review by chrismclarke — APPROVED** [review]
Thanks @khalifan-kfan - the issue with sharing as native file was indeed a tricky one. I noticed in the [Share Documentation](https://capacitorjs.com/docs/apis/share#android) that  by default only files stored in the `Cache` directory can be shared (without providing additional permissions), so I needed to add an extra method that allows copying to cache folder (see 6c6d579).

I also noticed play store links weren't sharing correctly (they only are stored as the app id so need extra part of the URL at the start to share), which was fixed in  56d5af3

Otherwise I just did a bit of code tidying and overall I think now should be good to go. Thanks for all your work on this!

**Pull request merged**